plugins/amongus/amongus.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "Game.h"
+
+#include "mumble_plugin.h"
+#include "mumble_plugin_utils.h"
+
+#include <sstream>
+
+std::unique_ptr< Game > game;
+
+static inline bool inGame(const AmongUsClient_Fields &fields) {
+	if (fields.gameMode != GameMode::FreePlay && fields.gameState != GameState::NotJoined) {
+		return true;
+	}
+
+	return false;
+}
+
+static int fetch(float *avatarPos, float *avatarFront, float *avatarTop, float *cameraPos, float *cameraFront,
+				 float *cameraTop, std::string &context, std::wstring &identity) {
+	for (uint8_t i = 0; i < 3; ++i) {
+		avatarPos[i] = avatarFront[i] = avatarTop[i] = cameraPos[i] = cameraFront[i] = cameraTop[i] = 0.f;
+	}
+
+	const auto fields = game->clientFields();
+
+	if (!inGame(fields)) {
+		return false;
+	}
+
+	if (fields.gameState == GameState::Ended) {
+		return true;
+	}
+
+	const auto playerControlFields = game->playerControlFields();
+
+	// Mumble | Game
+	// X      | X
+	// Y      | -
+	// Z      | Y
+	const auto position = game->playerPosition(playerControlFields.netTransform);
+	avatarPos[0] = cameraPos[0] = position.x;
+	avatarPos[2] = cameraPos[2] = position.y;
+
+	// Context
+	std::ostringstream stream;
+	stream << " {\"Game ID\": " << fields.gameId << "}";
+	context = stream.str();
+
+	stream.str(std::string());
+	stream.clear();
+
+	// Identity
+	const auto playerFields = game->playerInfoFields(playerControlFields.cachedData);
+
+	stream << "Name: " << game->string(playerFields.playerName) << std::endl;
```

Comment:
Using `std::endl` here doesn't make sense as every usage of it flushes the stream, which we don't need here. Thus I think it might be better to only use `"\n"` instead.